stripe/src/test/java/com/stripe/android/view/CountryTextInputLayoutTest.kt

```diff
@@ -136,6 +138,41 @@ class CountryTextInputLayoutTest {
         )
     }
 
+    @Test
+    fun `when screen rotates then selected country should carry over`() {
+        val application = ApplicationProvider.getApplicationContext<Application>()
+        application.setTheme(R.style.StripePaymentSheetDefaultTheme)
+        val oldCountryTextInputLayout = CountryTextInputLayout(
+            application
+        )
+
+        idleLooper()
+
+        assertEquals(CountryCode.US, oldCountryTextInputLayout.selectedCountryCode)
+        assertEquals("United States", oldCountryTextInputLayout.countryAutocomplete.text.toString())
+
+        oldCountryTextInputLayout.updatedSelectedCountryCode(CountryCode.CA)
+        oldCountryTextInputLayout.updateUiForCountryEntered(CountryCode.CA)
+
+        assertEquals(CountryCode.CA, oldCountryTextInputLayout.selectedCountryCode)
+        assertEquals("Canada", oldCountryTextInputLayout.countryAutocomplete.text.toString())
+
+        // mimic configuration change - the old instance's state is saved and passed to
+        // a new instance's onRestoreInstanceState.
+        // activityScenario.recreate() won't trigger onRestoreInstanceState
+        val oldState = oldCountryTextInputLayout.onSaveInstanceState()
+        val newCountryTextInputLayout = CountryTextInputLayout(
+            application
+        )
+        idleLooper()
+
+        // newCountryTextInputLayout.onResolvePointerIcon() is triggered during configuration change
```

Comment:
What is `onResolvePointerIcon`?

Reply:
sorry i meant `onRestoreInstanceState`, updated